sklearn/manifold/_t_sne.py

```diff
@@ -742,6 +749,12 @@ def _fit(self, X, skip_num_points=0):
                                    n_jobs=self.n_jobs,
                                    n_neighbors=n_neighbors,
                                    metric=self.metric)
+            if 'p' in metric_params_:
```

Comment:
is this handling necessary?

Reply:
We think so because the 'p' parameter is treated differently in `NearestNeightbors`: it is not part of 'metric_params' but a parameter that should be given separately. Do you think it should be done differently?